core/src/main/java/com/linecorp/armeria/server/HttpStatusException.java

```diff
@@ -32,51 +34,69 @@
 
     static {
         for (int i = 0; i < EXCEPTIONS.length; i++) {
-            EXCEPTIONS[i] = new HttpStatusException(HttpStatus.valueOf(i), false);
+            EXCEPTIONS[i] = new HttpStatusException(HttpStatus.valueOf(i), false, null);
         }
     }
 
     /**
      * Returns a new {@link HttpStatusException} instance with the specified HTTP status code.
      */
     public static HttpStatusException of(int statusCode) {
-        if (statusCode < 0 || statusCode >= 1000) {
-            final HttpStatus status = HttpStatus.valueOf(statusCode);
-            if (Flags.verboseExceptionSampler().isSampled(HttpStatusException.class)) {
-                return new HttpStatusException(status);
+        return of(HttpStatus.valueOf(statusCode));
+    }
+
+    /**
+     * Returns a new {@link HttpStatusException} instance with the specified HTTP status code and {@code cause}.
+     */
+    public static HttpStatusException of(int statusCode, Throwable cause) {
+        return of(HttpStatus.valueOf(statusCode), cause);
+    }
+
+    /**
+     * Returns a new {@link HttpStatusException} instance with the specified {@link HttpStatus}.
+     */
+    public static HttpStatusException of(HttpStatus status) {
+        final boolean sampled = Flags.verboseExceptionSampler().isSampled(HttpStatusException.class);
+        if (!sampled) {
+            final int statusCode = status.code();
+            if (statusCode >= 0 && statusCode < 1000) {
+                return EXCEPTIONS[statusCode];
             } else {
-                return new HttpStatusException(status, false);
+                return new HttpStatusException(HttpStatus.valueOf(statusCode), false, null);
             }
-        } else {
-            return EXCEPTIONS[statusCode];
```

Comment:
Did we fix a bug that normal status codes would never be `verbose`?

Reply:
Yes.

Reply:
Updated PR description